### PR TITLE
port allpay into oca repo.

### PR DESCRIPTION
**Pull request by jeffery9**

### CODE REVIEW

File: payment_allpay/README.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+# payment_allPay
```
Comment (elicoidal) [inline payment_allpay/README.md:1]: Hi @jeffery9 please use standard README.rst template (https://github.com/OCA/maintainer-tools/blob/master/template/module/README.rst)
There is no md file in OCA repos

File: payment_allpay/__init__.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
+# -*- coding: utf-8 -*-
```
Comment (elicoidal) [inline payment_allpay/__init__.py:1]: Please use simple headers for all py files following OCA templates:
https://github.com/OCA/maintainer-tools/blob/master/template/module/__openerp__.py#L3

File: payment_allpay/__openerp__.py [hunk 3]
```diff
@@ -0,0 +1,20 @@
+# -*- coding: utf-8 -*-
+
+{
+    'name': 'allPay Payment Acquirer',
+    'category': 'Website',
+    'summary': 'Payment Acquirer: allPay Implementation',
+    'version': '1.0',
+    'description': """allPay Payment Acquirer""",
```
Comment (elicoidal) [inline payment_allpay/__openerp__.py:8]: remove description not necesary (README.rst provides the information)

File: payment_allpay/__openerp__.py [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+    'author': 'Odoo CN, Jeffery <jeffery9@gmail.com>',
```
Comment (elicoidal) [inline payment_allpay/__openerp__.py:9]: follow https://github.com/OCA/maintainer-tools/blob/master/template/module/__openerp__.py and add Odoo Community Association as Author

File: payment_allpay/controllers/__init__.py [hunk 5]
```diff
@@ -0,0 +1,3 @@
+# -*- coding: utf-8 -*-
```
Comment (elicoidal) [inline payment_allpay/controllers/__init__.py:1]: add simple headerd everywhere

File: payment_allpay/models/allpay.py [hunk 6]
```diff
@@ -0,0 +1,251 @@
...
+        else:
+            percentage = acquirer.fees_int_var
+            fixed = acquirer.fees_int_fixed
+        fees = (percentage / 100.0 * amount + fixed ) / (1 - percentage / 100.0)
+        return fees
+
+    def allpay_form_generate_values(self, cr, uid, id, partner_values, tx_values, context=None):
+        base_url = self.pool['ir.config_parameter'].get_param(cr, SUPERUSER_ID, 'web.base.url')
+        acquirer = self.browse(cr, uid, id, context=context)
+        date_create = datetime.datetime.now().strftime('%Y/%m/%d %H:%M:%S')  # time format yyyy/MM/dd HH:mm:ss
```
Comment (elicoidal) [inline payment_allpay/models/allpay.py:152]: remove useless comment

File: payment_allpay/models/allpay.py [hunk 7]
```diff
@@ -0,0 +1,251 @@
...
+    # --------------------------------------------------
+
+    def _allpay_form_get_tx_from_data(self, cr, uid, data, context=None):
+        reference, txn_id = data.get('TradeNo'), data.get('MerchantTradeNo')
+        if not reference or not txn_id:
+            error_msg = 'allPay: received data with missing reference (%s) or txn_id (%s)' % (reference, txn_id)
+            _logger.error(error_msg)
+            raise ValidationError(error_msg)
+
+        # find tx -> @TDENOTE use txn_id ?
```
Comment (elicoidal) [inline payment_allpay/models/allpay.py:218]: The comment is quite cryptic: can you elaborate or remove?

**Comment by elicoidal** [comment]
@jeffery9 contribution looks good but need some cleansing on format and code formatting: please make sure to change all your code so that it is PEP8 compliant.
Once reformatted I will ask some technical reviewer to have a look.

One question: should not we move this PR to the tw repository (https://github.com/OCA/l10n-taiwan)?

**Comment by elicoidal** [comment]
@jeffery9 any feedback

**Comment by elicoidal** [comment]
@jeffery9 there has been no feedback in some time now. 
Are you going to fix the PR? do you need some help to get started?

**Comment by yelizariev** [comment]
@jeffery9 What is the license of these updates?

**Comment by yelizariev** [comment]
Ok, looks like it is AGPL

**Comment by jeffery9** [comment]
Yes,  AGPL.

**Pull request closed without merging**